core/services/job_subscriber.go

```diff
@@ -37,15 +37,32 @@ type jobSubscriber struct {
 	jobSubscriptions map[string]JobSubscription
 	jobsMutex        *sync.RWMutex
 	runManager       RunManager
+	headChannel      chan *models.Head
 }
 
 // NewJobSubscriber returns a new job subscriber.
 func NewJobSubscriber(store *store.Store, runManager RunManager) JobSubscriber {
-	return &jobSubscriber{
+	js := &jobSubscriber{
 		store:            store,
 		runManager:       runManager,
 		jobSubscriptions: map[string]JobSubscription{},
 		jobsMutex:        &sync.RWMutex{},
+		headChannel:      make(chan *models.Head, 1),
+	}
+	go js.jobLoop()
+	return js
+}
+
+// jobLoop is a long running goroutine which handles new heads
+func (js *jobSubscriber) jobLoop() {
+	for {
+		select {
+		case head := <-js.headChannel:
+			err := js.runManager.ResumeAllConfirming(head.ToInt())
+			if err != nil {
+				logger.Errorw("Failed to resume confirming tasks on new head", "error", err)
+			}
+		}
```

Comment:
would we want to support a graceful shutdown by exiting this loop?

Reply:
I think so 👍

Reply:
So a suggestion, which may be annoying and depends on how this entity is implemented, is add like methods for its lifecycle (Start(..) and Stop(..) in which stop closes js.headChannel)